test/integration/api/test_dist_analytics.py

```diff
@@ -1,10 +1,100 @@
+import os
+
+import pytest
+
 import pandas as pd
 from fastapi.testclient import TestClient
 
 from api.app.main import app
 
 client = TestClient(app)
 
+class TestDistAnalyticsWithNoPreviousResult:
```

Comment:
Ah I like this. So basically you can set up a test per each assertion but have each test be doing the same initial requests?

Reply:
Yes, this speeds things up significantly with one **very** important caveat. The tests that rely on the setup can't rely on each other. As long as we have test independence, all is well. For each of these tests, there is no further mutation of the payload or application state. We just read different parts of the response. So, it's okay for only one action to happen instead of one for each test.